DDOT_module/child_classes/correction_data_generator.py

```diff
@@ -0,0 +1,397 @@
+                        self.get_qubits_key(
+                            qubit_indices_for_construction[0]): averaged_matrices_cluster_i,
+                        self.get_qubits_key(
+                            qubit_indices_for_construction[1]): averaged_matrices_cluster_j
+                    }
+
+                    neighbors_for_construction = [[rev_map_enumerated[ci] for ci in
+                                                   intersection_i],
+                                                  [rev_map_enumerated[cj] for cj in
+                                                   intersection_j]]
+
+                    neighbors_for_construction_modified = []
+                    for index_subset in range(len(neighbors_for_construction)):
+                        subset_neighbors = neighbors_for_construction[index_subset]
+
+                        if len(subset_neighbors) == 0:
+                            neighbors_for_construction_modified.append(None)
+                        else:
+                            neighbors_for_construction_modified.append(subset_neighbors)
+
+                    neighbors_for_construction = neighbors_for_construction_modified
+
+                    big_lambda_creator_now = GlobalNoiseMatrixCreator(
+                        noise_matrices_dictionary=properly_formatted_lambdas)
+                    averaged_matrix_clusters_i_j = \
+                        big_lambda_creator_now.compute_global_noise_matrix(
+                            qubit_indices_for_construction,
+                            neighbors_for_construction
+                        )
+
+                    if calculate_mitigation_errors:
+                        # TODO: FIX THIS PART
+                        # qubits_clusters_ij = anf.get_enumerated_rev_map_from_indices(anf.lists_sum_multi([cluster_i,cluster_j]))
+                        qubits_ij = anf.get_enumerated_rev_map_from_indices(anf.lists_sum_multi(
+                            [cluster_i, cluster_j, dependencies_cluster_i,
+                             dependencies_cluster_j]))
+
+                        clu_i, clu_j = [qubits_ij[qc] for qc in cluster_i], [qubits_ij[qc] for qc
+                                                                             in cluster_j],
+                        deps_i, deps_j = [qubits_ij[qc] for qc in dependencies_cluster_i], [
+                            qubits_ij[qc] for qc in dependencies_cluster_j]
+
+                        all_qubit_ij = anf.lists_sum_multi([deps_i, deps_j])
+                        all_qubits_outside_ij = list(
+                            set(all_qubit_ij).difference(set(clu_i + clu_j)))
+
+                        possible_states_ij_outside = anf.register_names_qubits(
+                            range(len(all_qubits_outside_ij)), len(all_qubits_outside_ij))
+
+                        possible_states_ij = []
+                        for k in range(len(possible_states_ij_outside)):
```

Comment:
Wouldn't 
```suggestion
                        for state_here in possible_states_ij_outside:
```
do the same?